From 53800c1f8206e9988dff88a71915cda7e7bff6e3 Mon Sep 17 00:00:00 2001
From: Paul Wessel <pwessel@hawaii.edu>
Date: Sat, 4 Apr 2020 11:27:46 -1000
Subject: [PATCH] Check if netcdf4 grid before calling nc_inq_var_deflate

netCDF 4.7.4 released April 4, 2002 on macports caused massive crashes when reading netcdf3 grids s the nc_inq_var_deflate now suddenly crashes with an error instead of previous behavior.  Isolating this call by first checking if netcdf4.

[Homebrew edit: backported to 5.4 branch]
---
 src/gmt_nc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gmt_nc.c b/src/gmt_nc.c
index 7023561dd5a9103081c0ef6b4b0557c85bd6b8c4..02f4144b95a9f1c82c9515ba9d9cd9b457ec5c71 100644
--- a/src/gmt_nc.c
+++ b/src/gmt_nc.c
@@ -726,7 +726,7 @@ GMT_LOCAL int gmtnc_grd_info (struct GMT_CTRL *GMT, struct GMT_GRID_HEADER *head
 		}
 		{
 			/* get deflation and chunking info */
-			int storage_mode, shuffle, deflate, deflate_level;
+			int storage_mode, shuffle = 0, deflate = 0, deflate_level = 0;
 			size_t chunksize[5]; /* chunksize of z */
 			gmt_M_err_trap (nc_inq_var_chunking (ncid, z_id, &storage_mode, chunksize));
 			if (storage_mode == NC_CHUNKED) {
@@ -736,7 +736,7 @@ GMT_LOCAL int gmtnc_grd_info (struct GMT_CTRL *GMT, struct GMT_GRID_HEADER *head
 			else { /* NC_CONTIGUOUS */
 				header->z_chunksize[0] = header->z_chunksize[1] = 0;
 			}
-			gmt_M_err_trap (nc_inq_var_deflate (ncid, z_id, &shuffle, &deflate, &deflate_level));
+			if (header->is_netcdf4) gmt_M_err_trap (nc_inq_var_deflate (ncid, z_id, &shuffle, &deflate, &deflate_level));
 			header->z_shuffle = shuffle ? true : false; /* if shuffle filter is turned on */
 			header->z_deflate_level = deflate ? deflate_level : 0; /* if deflate filter is in use */
 		}
